### ISSUE TITLE
Build script should detect need to initialise git submodules

### ISSUE THREAD

**Issue by mcv21** [irods/irods#3270]
The build process for 4.2 is undocumented; there were a couple of dependencies I needed to install on ubuntu 12.04:

libpam0g-dev
unixodbc-dev

The error messages from cmake at least make this obvious, but documenting them would be nice.

But you also need to do `git submodule init && git submodule update`. If you don't, you get a pretty confusing error message:

```
CMake Error in CMakeLists.txt:
  Cannot find source file:

    /nfs/users/nfs_m/mv3/upstream/irods/irods_schema_messaging/v1/server_control_plane_command.json

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx
```

The cmake setup should notice the lack of submodules, and emit a helpful error message.

Additionally, these steps would be documented :)

---

Following added by @kellerb:
- [x] Helpful error message if messaging submodule not initialized
- [ ] Document non irods-externals externals, e.g. pam, odbc

**Comment by alanking** [comment]
For what it's worth, when I try to build main without `libpam0g-dev` on Ubuntu 20.04, I got the following message from CMake:
```
CMake Error at CMakeLists.txt:123 (message):
  pam library not found
```

So that's good, but it doesn't catch that I haven't installed `libkrb5-dev` (complains about not being able to find `gssapi.h`).

Dunno if that helps :)

**Comment by korydraughn** [comment]
The title of this issue mentions git submodules. The repo no longer relies on git submodules. See #6517.

Is `libkrb5-dev` still a required dependency for 4.3 given kerberos is not supported directly?

**Comment by korydraughn** [comment]
After some discussion, we've confirmed that the kerberos development package is required at this time.

Closing as resolved.

**Comment by alanking** [comment]
Removed checkbox:
- [ ] Document non irods-externals externals, e.g. pam, odbc

We can address these individually as they appear. For now, the ones we care about have been addressed (https://github.com/irods/irods/issues/3270#issuecomment-1010046069) or will be addressed (#7207) in other issues.
